packages/flutter/lib/src/material/tabs.dart

```diff
@@ -381,19 +381,61 @@ class _TabsScrollBehavior extends BoundedBehavior {
   }
 }
 
-class TabBarSelection {
-  TabBarSelection({ int index: 0, this.maxIndex, this.onChanged }) : _index = index {
+class _TabBarSelection extends InheritedWidget {
+  _TabBarSelection({
+    this.selection,
+    Key key,
+    Widget child
+  }) : super(key: key, child: child);
+
+  final TabBarSelectionState selection;
+
+  bool updateShouldNotify(_TabBarSelection oldWidget) => selection != oldWidget.selection;
```

Comment:
Yes, that's correct, the descendants just track the TabBarSelection's performance. The case I was trying to handle here was the (for now) unusual case where the inherited TabBarSelection is replaced or removed.